src/patches/dns.js

```diff
@@ -0,0 +1,39 @@
+// Lambdas running inside a VPC rely on ENIs. On a cold start attaching the ENI
+// can be quite slow and can mean that some services, like DNS, are not yet
+// functional when the Lambda function begins to execute for the first time.
+// Inserting DNS retry logic gives the function a chance to recover before
+// failing completely.
+//
+// See https://docs.aws.amazon.com/lambda/latest/dg/vpc.html#vpc-configuring
+(function () {
+  const dns = require('dns');
+  const lookup = dns.lookup;
+
+  const DELAY = 1000;
```

Comment:
In the context of most requests, this seems like a very long time to wait to retry a sns lookup. Should this be a smaller value, or perhaps configurable?

Reply:
I considered this. The thing is that ENI cold starts are quite slow so I don't expect that retrying more frequently is very likely to succeed. I'd rather not make this configurable as I think it adds undo complexity and I don't see any value in changing this parameter between projects.